File: functions/liboqs/variants/c/0_fpr_half_6.c
#include <stdint.h>

static uint64_t fpr_half(uint64_t x) {
    uint64_t subtractor = x ? (1ULL << 52) : 0;
    x -= subtractor;
    uint32_t exp = (uint32_t)(x >> 52);
    if (exp == 0) {
        return 0;
    }
    return x;
}


int main(){
    fpr_half(0);
}

File: functions/liboqs/variants/go/0_fpr_half_6.go
package main

func fpr_half(x uint64) uint64 {
    frac := x & ((1 << 52) - 1)
    exp := (x >> 52) & 0x7FF
    if exp == 0 {
        return x >> 1
    }
    exp--
    return frac | (exp << 52)
}


func main(){
    fpr_half(0)
}
